src/constraint.rs

```diff
@@ -1344,15 +1345,28 @@ impl AnnotationsConstraint {
 }
 
 impl ConstraintValidator for AnnotationsConstraint {
-    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+    fn validate(&self, value: &IonSchemaElement, type_store: &TypeStore) -> ValidationResult {
         let violations: Vec<Violation> = vec![];
 
-        // validate annotations that have list-level `ordered` annotation
-        if self.is_ordered {
-            return self.validate_ordered_annotations(value, type_store, violations);
+        match value {
+            IonSchemaElement::SingleElement(element) => {
+                // validate annotations that have list-level `ordered` annotation
+                if self.is_ordered {
+                    return self.validate_ordered_annotations(element, type_store, violations);
+                }
+
+                // validate annotations that does not have list-level `ordered` annotation
+                self.validate_unordered_annotations(element, type_store, violations)
+            }
+            IonSchemaElement::Document(document) => {
+                // document type can not have annotations
+                Err(Violation::new(
+                    "annotations",
+                    ViolationCode::AnnotationMismatched,
+                    "annotations don't match expectations",
```

Comment:
question—Should this be a "not applicable for this type" sort of violation instead? (Would that be clearer, do you think?)

Reply:
Yes, I think that is much precise. I will change it.